### ISSUE TITLE
MB_DB_CONNECTION_URI with percent encoding not working

### ISSUE THREAD

**Issue by looric** [metabase/metabase#7345]
metabase 0.28.5 
mysql database

export MB_DB_CONNECTION_URI=“mysql://metabase1:9ieZ83l@localhost:3306/metabase1?characterEncoding=utf-8”
is working as expected.
but db password like: Xz=/9ieZ83=TBPK-I{p,Ki=l@f$J22<l
not working with Percent encoding 
export MB_DB_CONNECTION_URI="mysql://metabase:Xz%3D%2F9ieZ83%3DTBPK-I%7Bp%2CKi%3Dl%40f%24J22%3Cl@localhost:3306/metabase?characterEncoding=utf-8"

**Comment by mijamo** [comment]
This is particularly needed for instance when you want to run metabase on Google Cloud Run because in that case the SQL instance is accessible through a socket `/cloudsql/project:region:instance` and there is currently no way to make that work because the `:` breaks the regex and the percent encoded value is not decoded

**Comment by flamber** [comment]
This is already supported by moving the password to the parameters - and I think that's the JDBC "standard":
```
mysql://metabase@localhost:3306/metabase?password=Xz%3D%2F9ieZ83%3DTBPK-I%7Bp%2CKi%3Dl%40f%24J22%3Cl&characterEncoding=utf-8
```

**Comment by revolunet** [comment]
This also happen on Azure Postgres instances which have the default connection string like this :

`username%40hostname:password@hostname/database?ssl=require`

**Comment by camsaul** [comment]
This should be fixed by #14275
